Update docs with the officially supported downgrade mechanism (#57266)
gravitational/teleport: docs/pages/upgrading/overview.mdx

1. Verify the upgrade was successful.
1. Upgrade the trusted leaf clusters.

> ## Downgrading Major Versions

Reviewer: ```suggestion
## Downgrading major versions
```
We use sentence case in headings

In order to safely downgrade from one major version to another, i.e. from v17 to v16,
you must restore the Teleport backend to the back up taken prior to upgrading. First,

> rollback all components to the exact version they were running prior to the upgrade in

Reviewer: ```suggestion
roll back all components to the exact version they were running prior to the upgrade in
```
"roll back" is the verb, while "rollback" is the noun

> reverse order of the upgrade sequence above until only the Auth instances are running

Reviewer: ```suggestion
reverse order of the upgrade sequence above until only the Auth Service instances are running
```
For consistent component naming

> the new major version. Then stop the Auth service and follow the

Reviewer: ```suggestion
the new major version. Then stop the Auth Service and follow the
```

[Backup and Restore](../admin-guides/management/operations/backup-restore.mdx) guidance
to restore the backend to a point in time prior to the upgrade. Once the backup has been

> restored, downgrade the Auth instances to the exact version of Teleport

Reviewer: ```suggestion
restored, downgrade the Auth Service instances to the exact version of Teleport
```

> restored, proceed to downgrading the Auth instances to the exact version of Teleport

Reviewer 2: ```suggestion
restored, downgrade the Auth instances to the exact version of Teleport
```

> they were running prior to the upgrade attempt.

Reviewer 2: ```suggestion
they were running prior to the upgrade attempt, then start them again.
```
